app/src/test/java/org/mozilla/fenix/wallpapers/WallpaperDownloaderTest.kt

```diff
@@ -0,0 +1,134 @@
+            url = "$remoteHost/${wallpaper.collection.name}/${wallpaper.name}/portrait.png",
+            method = Request.Method.GET
+        )
+        val landscapeRequest = Request(
+            url = "$remoteHost/${wallpaper.collection.name}/${wallpaper.name}/landscape.png",
+            method = Request.Method.GET
+        )
+        every { mockPortraitResponse.status } returns 200
+        every { mockLandscapeResponse.status } returns 200
+        every { mockPortraitResponse.body } returns portraitResponseBodySuccess
+        every { mockLandscapeResponse.body } returns landscapeResponseBodySuccess
+        every { mockClient.fetch(portraitRequest) } returns mockPortraitResponse
+        every { mockClient.fetch(landscapeRequest) } returns mockLandscapeResponse
+
+        downloader.downloadWallpaper(wallpaper)
+
+        val expectedPortraitFile = File(tempFolder.root, "wallpapers/${wallpaper.name}/portrait.png")
+        val expectedLandscapeFile = File(tempFolder.root, "wallpapers/${wallpaper.name}/landscape.png")
+        assertTrue(expectedPortraitFile.exists() && expectedPortraitFile.readText() == wallpaperBytes)
+        assertTrue(expectedLandscapeFile.exists() && expectedLandscapeFile.readText() == wallpaperBytes)
+    }
+
+    @Test
+    fun `GIVEN that request fails WHEN downloading THEN file is not created`() = runTest {
+        val wallpaper = generateWallpaper()
+        val portraitRequest = Request(
+            url = "$remoteHost/${wallpaper.collection.name}/${wallpaper.name}/portrait.png",
+            method = Request.Method.GET
+        )
+        val landscapeRequest = Request(
+            url = "$remoteHost/${wallpaper.collection.name}/${wallpaper.name}/landscape.png",
+            method = Request.Method.GET
+        )
+        every { mockPortraitResponse.status } returns 400
+        every { mockLandscapeResponse.status } returns 400
+        every { mockClient.fetch(portraitRequest) } returns mockPortraitResponse
+        every { mockClient.fetch(landscapeRequest) } returns mockLandscapeResponse
+
+        downloader.downloadWallpaper(wallpaper)
+
+        val expectedPortraitFile = File(tempFolder.root, "wallpapers/${wallpaper.name}/portrait.png")
+        val expectedLandscapeFile = File(tempFolder.root, "wallpapers/${wallpaper.name}/landscape.png")
+        assertFalse(expectedPortraitFile.exists())
+        assertFalse(expectedLandscapeFile.exists())
+    }
+
+    @Test
+    fun `GIVEN that copying the file fails WHEN downloading THEN file is not created`() = runTest {
+        val wallpaper = generateWallpaper()
+        val portraitRequest = Request(
+            url = "$remoteHost/${wallpaper.collection.name}/${wallpaper.name}/portrait.png",
+            method = Request.Method.GET
+        )
+        val landscapeRequest = Request(
+            url = "$remoteHost/${wallpaper.collection.name}/${wallpaper.name}/landscape.png",
+            method = Request.Method.GET
+        )
```

Comment:
if these requests are the same for all the tests, you could optionally make a `generateXRequest()` helper function like you did with `generateWallpaper()`

Reply:
🤦 yeah of course. why did I do that to myself lmao. thanks!